hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -412,13 +404,25 @@ public long tryReserveForUpdate(K key, Data keyData) {
             return NOT_RESERVED;
         }
 
-        R reservedRecord = getOrCreateToReserve(key, keyData);
         long reservationId = nextReservationId();
-        if (reservedRecord.casRecordState(RESERVED, reservationId)) {
-            return reservationId;
-        } else {
+        R reservedRecord = getOrCreateToReserve(key, keyData, reservationId);
+        return reservedRecord.getReservationId() == reservationId ? reservationId : NOT_RESERVED;
+    }
+
+    @Override
+    public long tryReserveForCacheOnUpdate(K key, Data keyData) {
+        checkAvailable();
+        // if there is no eviction configured we return if the Near
+        // Cache is full and it's a new key (we have to check the
+        // key, otherwise we might lose updates on existing keys)
+        if (evictionDisabled && evictionChecker.isEvictionRequired() && !containsRecordKey(key)) {
             return NOT_RESERVED;
         }
+
+        long reservationId = nextReservationId();
+        R reservedRecord = reserveForCacheOnUpdate(key, keyData, reservationId);
```

Comment:
This is the only different line between the newly added method `tryReserveForCacheOnUpdate` and the existing `tryReserveForUpdate`. We can actually combine both by using a `Supplier<R>` to the method and hence no need for the new method.

Reply:
unified them